src/package.rs

```diff
@@ -352,8 +340,24 @@ impl TryFrom<Bytes> for Package {
 pub enum PackageType {
     /// A library package containing primitive type definitions
     Lib,
-    /// A api package containing message and service definition
+    /// An api package containing message and service definition
     Api,
+    /// An implementation package that implements an api or library
+    ///
+    /// Note: Implementation packages can't be published via Buffrs
+    Impl,
```

Comment:
sorta, i introduced it to create a symmetry which makes the impl internally way easier as you otherwise have to check "package is none" instead of "package is an impl"